### PR TITLE
Feature/edit character

### PR DESCRIPTION
**Pull request by antgreen13**
Add functionality to edit a selected character

This change will set up a user interaction to allow a user to edit one of their selected characters.

### CODE REVIEW

File: src/app/creation/creation.component.ts [hunk 1]
```diff
@@ -19,8 +19,8 @@ import { ApiService } from '../api.service';
 })
 export class CreationComponent implements OnInit {
   characterName: string;
-  selectedRace: Race;
-  selectedClass: Class;
+  selectedRace: Race | undefined;
+  selectedClass: Class | undefined;
```
Comment (ericchen94) [inline src/app/creation/creation.component.ts]: Can make properties optional with `?:`

File: src/app/creation/creation.component.ts [hunk 2]
```diff
@@ -65,9 +97,9 @@ export class CreationComponent implements OnInit {
   }
 
   public saveCharacter() {
-    this.character.name = this.characterName;
-    this.character.race = this.selectedRace.name;
-    this.character.class = this.selectedClass.name;
+    this.character.name = this.characterName ? this.characterName : '';
```
Comment (ericchen94) [inline src/app/creation/creation.component.ts]: nit: you can use a logical OR here:
`this.character.name = this.characterName || '';`

File: firebase_test.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        character = json.loads(response.text)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(character, self.character)
+    
+    @mock.patch('app.firebase')
+    def test_delete_character(self, mock_firebase):
+        mock_firebase.return_value = 200
+        endpoint = f'/firebase/delete?userEmail={self.userEmail}&characterName={self.characterName}'
+        response = self.test_app.delete(endpoint)
+        self.assertEqual(response.status_code, 200)
```
Comment (ericchen94) [inline firebase_test.py:59]: Good practice to add tests to check for failures too, such as getting the expected error response code when running an invalid endpoint, etc

**Review by ericchen94 — CHANGES_REQUESTED** [review]

File: firebase_test.py (hunk 3 above)
Comment (antgreen13) [reply firebase_test.py:59]: Good idea. I'll be adding more tests as part of my current work, so I'll add those in then.

**Review by ericchen94 — APPROVED** [review]

**Pull request merged**